src/library_dylink.js

```diff
@@ -67,7 +67,15 @@ var LibraryDylink = {
       '__cpp_exception',
       '__wasm_apply_data_relocs',
       '__dso_handle',
-      '__set_stack_limits'
+      '__set_stack_limits',
+      '__clang_call_terminate',
+#if USE_PTHREADS
+      '__wasm_init_memory_flag',
+      '__wasm_init_memory',
+      '__wasm_init_tls',
+      '__tls_size',
+      '__tls_align',
+#endif
```

Comment:
are all these needed?

Reply:
They (slightly) help reduce the number of warnings about duplicated symbols when building with `ASSERTIONS`.
In practice lots of warnings are still shown due to what should be weak exports (e.g., C++ template instantiations).
None of them is necessary to fix the abort. I don't feel strongly about keeping any of them in this PR.

Reply:
That seems like a separate issue - I'm not sure why `ASSERTIONS=2` would cause any changes to missing symbols warnings or errors. Perhaps file an issue and we'll discuss and investigate there?